ui/components/component-library/button-primary/button-primary.types.ts

```diff
@@ -0,0 +1,39 @@
+import type { ButtonBaseStyleUtilityProps } from '../button-base/button-base.types';
+import type { PolymorphicComponentPropWithRef } from '../box';
+
+export enum ButtonPrimarySize {
+  Sm = 'sm',
+  Md = 'md',
+  Lg = 'lg',
+}
+
+export type ValidButtonTagType = 'button' | 'a';
+export interface ButtonPrimaryStyleUtilityProps
+  extends Omit<ButtonBaseStyleUtilityProps, 'size'> {
+  /**
+   * Boolean to change button type to Danger when true
+   */
+  danger: boolean;
+  /**
+   * Boolean to disable button
+   */
+  disabled: boolean;
+  /**
+   * Boolean to show loading spinner in button
+   */
+  loading: boolean;
+  /**
+   * Possible size values: 'ButtonPrimarySize.Auto'(auto), 'ButtonPrimarySize.Sm'(32px), 'ButtonPrimarySize.Md'(40px), 'ButtonPrimarySize.Lg'(48px), 'ButtonPrimarySize.Inherit'(inherits parents font-size)
```

Comment:
Nit: Could this be formatted so it's easier to read?

Reply:
We list `Auto` and `Inherit` here as sizes but they aren't included in the enum. We could also add these sizes to the enums so that on hover you can see the equivalent pixel sizes. More overhead to keep our docs up to date but I think it would improve the developer experience 


![Screenshot 2023-08-03 at 3 17 18 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/c86abcd0-f71a-418e-aaab-c6a168a39bb6)